src/parser/parsers.h

```diff
@@ -675,6 +691,23 @@ MaybeResult<typename Ctx::ContinuationT> conttype(Ctx& ctx) {
   return ctx.makeContType(*x);
 }
 
+// handlertype ::= '(' 'handler'  t*:vec(param) ')' => handler [t*]
+template<typename Ctx>
+MaybeResult<typename Ctx::HandlerT> handlertype(Ctx& ctx) {
+  if (!ctx.in.takeSExprStart("handler"sv)) {
+    return {};
+  }
+
+  auto elems = ctx.makeTupleElemList();
+  while (!ctx.in.takeRParen()) {
+    auto type = valtype(ctx);
+    CHECK_ERR(type);
+    ctx.appendTupleElem(elems, *type);
+  }
```

Comment:
for consistency with the reference interpreter and `wasmparser`, there should be the `result` keyword here too. that is, `(handler (result i32 i64 ... ))`

Reply:
Good catch! Thanks.